tests/Unit/TestHelpers.hpp

```diff
@@ -91,6 +91,8 @@ T serialize_and_deserialize(const T& t) {
 }
 
 /// Test for copy semantics assuming operator== is implement correctly
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wparentheses"
```

Comment:
Catch is doing something extremely clever in order to get the information for its warning messages, and that is triggering the warning.  (It is trying to do the pragmas itself as well.  I don't know why that isn't working.)  But there is nothing we can do in our code to satisfy the compiler beyond just silencing the warning.

Details:
The relevant bit from catch is the subexpression `__catchResult <= expr`.  (`__catchResult` is some internal catch type.) `expr` is the macro argument, so it expands to something like `__catchResult <= foo == bar`, which, according to precedence rules, is evaluated as `(__catchResult <= foo) == bar`.  So you can see that there's no way we can parenthesize the thing the compiler wants us to, because we only supply half of it.